server/controllers/LikesDislikeController.js

```diff
@@ -0,0 +1,233 @@
+// import TokenManager from '../helpers/TokenManager';
+import db from '../models';
+
+const { Article, ArticleLikesDislike } = db;
+
+/**
+ * @class LikesDislikeController
+ */
+class LikesDislikeController {
+  /**
+   * @description Method to like and unlike an article
+   * @static
+   * @param {*} req Request parameter
+   * @param {*} res Response parameter
+   * @returns {object} Json response
+   * @memberof LikesDislikeController
+   */
+  static async likeArticle(req, res) {
+    try {
+      const articleSlug = req.params;
+      const theArticleSlug = JSON.stringify(articleSlug).split('"')[3];
+      // UserId belongs to user who initiates like
+      const theArticle = await Article.findOne({
+        where: { slug: theArticleSlug }
+      });
+
+      if (!theArticle) {
+        return res.status(404).json({
+          data: { status: 'failure', message: 'Article Not Found' }
+        });
+      }
+      const unlikeTheArticle = await ArticleLikesDislike.findOne({
+        where: {
+          userId: req.user.userId,
+          articleId: theArticle.dataValues.id,
+          reaction: 'like'
+        }
+      });
+
+      if (unlikeTheArticle) {
+        await ArticleLikesDislike.destroy({
+          where: {
+            userId: req.user.userId,
+            articleId: theArticle.dataValues.id,
+            reaction: 'like'
+          }
+        });
+        return res.status(200).json({
+          data: { status: 'success', message: 'You just unliked this article!' }
+        });
+      }
+
+      const likeTheArticle = await ArticleLikesDislike.create({
+        userId: req.user.userId,
+        articleId: theArticle.dataValues.id,
+        reaction: 'like'
+      });
+
+      if (likeTheArticle) {
+        return res.status(200).json({
+          data: { status: 'success', message: 'You just liked this article!' }
+        });
+      }
+    } catch (error) {
+      return res.status(401).json({
+        data: { status: 'failure', message: error }
+      });
+    }
+  }
+
+  /**
+   * @description Method to dislike and undislike an article
+   * @static
+   * @param {*} req Request parameter
+   * @param {*} res Response parameter
+   * @returns {object} Json response
+   * @memberof LikesDislikeController
+   */
+  static async dislikeArticle(req, res) {
+    try {
+      const articleSlug = req.params;
+      const theArticleSlug = JSON.stringify(articleSlug).split('"')[3];
+      // UserId belongs to user who initiates like
```

Comment:
In my opinion, there's no need for this inline comment. Jsdoc is enough

Reply:
Resolved